BookWebApi/Service/BookService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+using BookWebApi.Model;
+using BookWebApi.DAO;
+using BookWebApi.CustomException;
+using BookWebApi.Util;
+
+namespace BookWebApi.Service
+{
+    public class BookService : IBookService
+    {
+        private BookData bookData = new BookData();
+
+        public List<Book> GetBook()
+        {
+            return bookData.Get();
+        }
+
+        public Book GetBook(int id)
+        {
+            if (id < 0)
+            {
+                throw new IncorrectIdException("Invalid Id, Id should be a positive number.");
+            }
+            Book book = bookData.GetById(id);
+            if (book == null)
+            {
+                throw new IncorrectIdException("No Book with id " + id + ".");
+            }
+            return book;
+        }
+
+        public Book CreateBook(Book book)
+        {
+            if (book.id < 0)
+            {
+                throw new IncorrectIdException("Invalid Id, Id should be a positive number.");
```

Comment:
You can append multiple error messages here. Instead of throwing exception from here, you can append multiple status codes and error messages to the a response object and then send that at the end of the function